app/cdap/components/CreateAndEditRecipeForm/__test__/CreateAndEditRecipeForm.test.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright © 2022 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render, screen, fireEvent } from '@testing-library/react';
+import React from 'react';
+import T from 'i18n-react';
+import CreateAndEditRecipeForm from 'components/CreateAndEditRecipeForm/index';
+import MyDataPrepApi from 'api/dataprep';
+
+describe('Test Create and Edit Recipe Component', () => {
+  const mockCreateAndEditRecipe = jest.fn();
+
+  jest.spyOn(MyDataPrepApi, 'updateRecipe').mockImplementation(() => {
+    return {
+      subscribe: (callback) => {
+        callback();
+      },
+    };
+  });
+
+  beforeEach(() => {
+    render(
+      <CreateAndEditRecipeForm
+        recipeData={{
+          recipeName: 'Abhilash',
+          description: 'Batman',
+          directives: [],
```

Comment:
For the sake of completeness, can you add a directive here?

Reply:
Done.